app/grandchallenge/evaluation/models.py

```diff
@@ -1403,6 +1407,35 @@ def metrics_json_file(self):
             if output.interface.slug == "metrics-json-file":
                 return output.value
 
+    @property
+    def missing_metrics_json_file(self):
+        if self.status == self.SUCCESS:
+
+            output_values = [
+                o.value
+                for o in self.outputs.all()
```

Comment:
Wouldn't it be more efficient to filter in the database (doing `self.outputs.filter(interface__slug="metrics-json-file")` rather than retrieving all and then filtering in python?

Reply:
If you do this, you would need to catch the case where there are mutliple outputs of this kind. Then you don't need to use the custom `get()` method later on.